src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -380,7 +380,16 @@ private static string GenerateUpgradeReport(string? projectName, string backupPa
                     TelemetryServiceUtility.StartOrResumeTimer();
 
                     IReadOnlyList<ProjectAction> actions = await resolveActionsAsync(projectManagerService);
-                    IReadOnlyList<PreviewResult> results = await GetPreviewResultsAsync(projectManagerService, actions, userAction, uiService, cancellationToken);
+                    IReadOnlyList<PreviewResult> results;
+                    try
+                    {
+                        results = await GetPreviewResultsAsync(projectManagerService, actions, userAction, uiService, cancellationToken);
+                    }
+                    catch
+                    {
+                        status = NuGetOperationStatus.Failed;
```

Comment:
Looking for alternative suggestions. The goal is to bail out of the action if any of the `NU1110`'s occur.